orchestration/dagster_orchestration/hca_manage/check.py

```diff
@@ -38,12 +48,251 @@ def check_data(args: argparse.Namespace, host: str, parser: argparse.ArgumentPar
     else:
         project = args.project
 
-    hca = HcaManage(environment=args.env,
-                    project=project,
-                    dataset=args.dataset,
-                    data_repo_client=get_api_client(host=host))
+    hca = CheckManager(environment=args.env,
+                       project=project,
+                       dataset=args.dataset,
+                       data_repo_client=get_api_client(host=host))
 
     if remove:
         hca.remove_all()
     else:
         hca.check_for_all()
+
+
+@dataclass
+class CheckManager(SoftDeleteManager):
```

Comment:
i agree with this thought and want to expand on it - this class feels like it was decomposed by topic (i.e. "stuff to check and validate rows") instead of by purpose ("fetching rows," "validating rows," "marking rows," etc). it's handling fetching records, running validations on them, and marking them as soft-deleted, as well as utility functions like collecting values into a CSV and running queries in BigQuery. i think we need to work out what individual things this class does and separate out functionality in addition to purpose.

Reply:
Hmm there are a bunch of attributes that they share so I figured it'd be simpler to just inherit and extend. I'm open to changing it but this was simple and maintained most of the code in a copy-pastable format haha

Reply:
I'd like to keep these classes focused and ensure a good separation of concerns, if possible. It's hard to eyeball this, which attributes are shared?

Reply:
usually if i find two objects share a lot of attributes but one's purpose/functionality isn't a superset of the other's, it's a sign i should pull out the common properties into their own class and have both inherit from that new class.

Reply:
If we decompose this by purpose rather than topic, do we still have a topic-based object that is simply composed of the purpose-based ones? I think it is useful to organize by "this is how we think of each separate tool", so therefore I think it is useful to have a representation of "this is the thing that does the checks we want". I think you both have more evolved thoughts than I on this though and I am very happy to learn why we should do this; I simply want to make sure I understand it 🙂

Reply:
topic-based classes are definitely valuable for command-line tools, and in my mind, should be very lightweight mappings between CLI arguments/flags and pre-existing functionality in more purpose-focused classes. if you've worked in an MVC framework before, think of the topic-based class as the controller and the purpose/data-based class as the model.